src/Compilers/Core/CommandLine/BuildProtocol.cs

```diff
@@ -45,17 +45,17 @@ namespace Microsoft.CodeAnalysis.CommandLine
     /// </summary>
     internal class BuildRequest
     {
-        public readonly uint ProtocolVersion;
+        public readonly Guid RequestId;
         public readonly RequestLanguage Language;
         public readonly ReadOnlyCollection<Argument> Arguments;
         public readonly string CompilerHash;
 
-        public BuildRequest(uint protocolVersion,
-                            RequestLanguage language,
+        public BuildRequest(RequestLanguage language,
                             string compilerHash,
-                            IEnumerable<Argument> arguments)
+                            IEnumerable<Argument> arguments,
+                            Guid? requestId = null)
         {
-            ProtocolVersion = protocolVersion;
+            RequestId = requestId ?? Guid.Empty;
```

Comment:
What are the times we'd expect the request id to be empty?

Reply:
When the build is invoked via test code. There isn't much value in tracking the requests there that I could see.